doc: nrf: migration: Add note on MCUboot picolibc change (#24584)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/migration/migration_guide_3.2.rst

This section describes the changes related to samples and applications.

MCUboot
-------

> The default C library for MCUboot has changed to picolibc which is recommended over the minimal C library as it is a fully developed and supported C library designed for application usage.

Author: Fixed

This section describes the changes related to samples and applications.

MCUboot
-------

The default C library for MCUboot has changed to picolibc.
Picolibc is recommended over the minimal C library as it is a fully developed and supported C library designed for application usage.

> If the C library for MCUboot in a sysbuild project has not been explicitly changed by using either a :file:`sysbuild/mcuboot/prj.conf` file, or :file:`sysbuild/mcuboot.conf` file, it will automatically be set to use picolibc.

Reviewer: ```suggestion
If you have not explicitly specified the C library in your sysbuild project for MCUboot using either a :file:`sysbuild/mcuboot/prj.conf` file or :file:`sysbuild/mcuboot.conf` file, picolibc will be used by default.
```

Author: Fixed

> Use :kconfig:option:`CONFIG_PICOLIBC` to specify piclibc in a project or :kconfig:option:`CONFIG_MINIMAL_LIBC` to specify the minimal C library (not recommended for usage outside of tests).

Reviewer: ```suggestion
To set picolibc in your project, use the :kconfig:option:`CONFIG_PICOLIBC` Kconfig option. 
If you need to use the minimal C library (which is not recommended outside of testing scenarios), use the :kconfig:option:`CONFIG_MINIMAL_LIBC` Kconfig option.
```
